src/client/util/browser.js

```diff
@@ -0,0 +1,61 @@
+function getMSEdgeVersion (userAgent) {
+    var edgeStrIndex = userAgent.indexOf('edge/');
+
+    return parseInt(userAgent.substring(edgeStrIndex + 5, userAgent.indexOf('.', edgeStrIndex)), 10);
+}
+
+function calculateBrowserAndVersion (userAgent) {
+    var webkitRegEx  = /(webkit)[ \/]([\w.]+)/;
+    var operaRegEx   = /(opera)(?:.*version)?[ \/]([\w.]+)/;
+    var msieRegEx    = /(msie) ([\w.]+)/;
+    var firefoxRegEx = /(firefox)/;
+
+    var match = webkitRegEx.exec(userAgent) ||
+                operaRegEx.exec(userAgent) ||
+                msieRegEx.exec(userAgent) ||
+                userAgent.indexOf('compatible') < 0 && firefoxRegEx.exec(userAgent) ||
+                [];
+
+    return {
+        name:    match[1] || '',
+        version: match[2] || '0'
+    };
+}
+
+export function init (win) {

```

Comment:
Why we need `init` here? Browser info is the same for all windows, so just export vars without init call.